text/0000-export-to-oci.md

```diff
@@ -0,0 +1,343 @@
+ | `<oci-dir>` | `CNB_OCI_PATH` | /oci | Path to oci directory where the images are saved |
+
+- WHEN `the new flag -oci or the default environment variable CNB_USE_OCI are set to true` during the phase invocation THEN the feature will be enabled.
+The image look up will be done following these rules:
+  - WHEN `the image points to a tag reference`
+    - Lifecycle will load the image from disk in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format at `<oci-dir>/<registry>/<repo>/<tag>`
+  - WHEN `the image points to a digest reference`
+    - Lifecycle will load the image from disk in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format at `<oci-dir>/<registry>/<repo>/<digest>`
+  - Apart from the look up, the logic for each phase should remain the same
+
+## Proof of concept
+
+In order to validate the feasibility of the proposed feature, we developed a proof of concept with one of the most important side effects this capability can add into the project: **Removing the Daemon Support**. You can also check a recording with the demo in the following [link](https://drive.google.com/file/d/1W1125OHuyUlx88BRroUTLBfrFHhFM5A9/view?usp=sharing)
+
+As mentioned earlier, if we want to remove the daemon support in the Lifecycle, then all the responsibility to deal with it goes into the platforms implementors, that means, for example:
+- Pull the require dependencies (runtime image for example), save them on disk in OCI layout format and pass it through the lifecycle using the `<oci-dir>` parameter
+- Push the application image (exported in OCI layout format) into the Daemon, because that is what users are expecting.
+
+During the proof of concept implementation I choose to use [skopeo](https://github.com/containers/skopeo) tool to solve the problem of interacting with the Daemon. The reason to do it was **simplicity** for the PoC developed but we believe this is a good subject to talk about with the community.
+
+The following workflow was developed:
+- Pack download the [skopeo image](https://github.com/containers/skopeo/blob/main/install.md#container-images), similar as it is downloading the other dependencies (Lifecycle, Buildpacks)
+- Pack executes [skopeo copy](https://github.com/containers/skopeo/blob/main/docs/skopeo-copy.1.md)  command in a container
+  - Copy image from the Daemon into the filesystem, in OCI layout format, before running Lifecycle
+  - Copy image from filesystem into the Daemon after the export phase was executed
+
+The following Dynamic Diagram from the C4 model, can give a little idea of the pieces implemented during the Poc
+
+![](https://i.imgur.com/SkY3l62.png)
+
+### Measuring of performance impact
+
+In order to have an idea on how much is affected the performance of exporting to the Daemon using the OCI layout format, the following metrics were taken.
+
+Using a local workstation with the following specifications:
```

Comment:
Can we get a linux workstation setup in a cloud to run these tests on? The disk virtualization layer in macOS is known for being pretty slow in certain situations. I'm curious if OCI as the default would be much faster on linux with docker daemon.

Reply:
Well, I have a very good personal machine running Pop OS, I will run there! thanks for the idea!